manager/manager.go

```diff
@@ -769,11 +769,35 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
 	return nil
 }
 
+// As we are proposing having multiple ways to detect a container (in addition to cgroups)
+// need the ability to delete
```

Comment:
so currently we detect containers via raw cgroups (what I'll call the "raw" method).  I'm proposing the ability to detect containers via other methods, such as the rkt api service.

if a raw handler was already created for a cgroup path for a rkt container (via raw detection), we want the rkt path to overwrite/update it to the rkt handler.  Or at least that's what I thought you wanted.  i.e. the raw handler will be destroyed and a rkt handler will take its place.  all in one function to ensure atomic nature.

if the rkt handler is created first, the raw handler wont be created as createContainer() skips if the cgroup is already a key in the map.